tenseal/cpp/tensors/plain_tensor.h

```diff
@@ -41,13 +55,45 @@ class PlainTensor {
         }
 
         _shape = {H, W};
-        _strides = {W, 1};
+        _strides = generate_strides(_shape);
         _data.reserve(H * W);
 
         for (auto& vec : data) {
             _data.insert(_data.end(), vec.begin(), vec.end());
         }
     }
+    /**
+     * Create a new PlainTensor from an ND vector.
+     * @param[in] input vector.
+     * @param[in] input shape.
+     */
+    PlainTensor(const vector<plain_t>& data, const vector<size_t>& shape)
+        : _data(data), _shape(shape), _strides(generate_strides(shape)) {
+        size_t expected_size = 1;
+        for (auto& d : shape) expected_size *= d;
+        if (data.size() != expected_size)
+            throw invalid_argument("tensor with mismatched shape");
+    }
+    /**
+     * Create a new PlainTensor from a batched tensor.
+     * @param[in] input matrix.
+     * @param[in] original shape.
+     * @param[in] batching axis.
+     */
+    PlainTensor(const vector<vector<plain_t>>& data,
+                const vector<size_t>& shape, size_t dim)
+        : _shape(shape), _strides(generate_strides(shape)) {
+        if (data[0].size() != shape[dim])
+            throw invalid_argument("invalid dimension shape");
+
+        _data.resize(data.size() * data[0].size());
+
+        for (size_t batch_idx = 0; batch_idx < data.size(); ++batch_idx) {
+            for (size_t idx = 0; idx < data[batch_idx].size(); ++idx) {
+                _data[idx * data.size() + batch_idx] = data[batch_idx][idx];
+            }
+        }
+    }
```

Comment:
The same, the shape/reshape directly impacts the strides, no? We can recreate the strides whenever we need them, from the shape, instead of storing them

Reply:
it will depends on how the reshape function works, I think that one advantage of using this strided representation is to reduce data movement, but only update the strides.

Reply:
We have this representation for now because we don't have reshape or other ops that change the shape after tensor construction.

We can recreate the strides everytime they're needed, they are not that big to cause performance issues.